mysql-test/suite/rocksdb/t/collation.test

```diff
@@ -166,16 +96,76 @@ CREATE TABLE c (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rock
 DROP TABLE a, b;
 
 # test invalid regex (trailing escape)
---exec echo "" >$MYSQLTEST_VARDIR/log/mysqld.1.err
 SET GLOBAL rocksdb_strict_collation_exceptions="abc\\";
---exec grep "Invalid pattern" $MYSQLTEST_VARDIR/log/mysqld.1.err | cut -d] -f2
+--let SEARCH_PATTERN=Invalid pattern in strict_collation_exceptions: abc\\\\
+--source include/search_pattern_in_file.inc
+
 --error ER_UNKNOWN_ERROR
 CREATE TABLE abc (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+--error ER_UNKNOWN_ERROR
+CREATE TABLE ABC (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+
 SET GLOBAL rocksdb_strict_collation_exceptions="abc";
 CREATE TABLE abc (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+DROP TABLE abc;
+--error ER_UNKNOWN_ERROR
+CREATE TABLE ABC (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+
+SET GLOBAL rocksdb_strict_collation_exceptions="ABC";
+--error ER_UNKNOWN_ERROR
+CREATE TABLE abc (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+CREATE TABLE ABC (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+DROP TABLE ABC;
+
+--error ER_UNKNOWN_ERROR
+CREATE TABLE bcd (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+
+SET GLOBAL rocksdb_strict_collation_exceptions="^ABC";
 --error ER_UNKNOWN_ERROR
 CREATE TABLE abcd (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+CREATE TABLE ABCD (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+DROP TABLE ABCD;
+
+--error ER_UNKNOWN_ERROR
+CREATE TABLE ZABC (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+
+
+# test with server comparing in lower case
+--let $restart_parameters="restart: --lower_case_table_names=1"
```

Comment:
George, I don't think this is nearly as bad as you'd seem to see. This is simply about testing the interaction of two server variables, which does have different behavior depending on the settings. Were this random two other variables, we'd simply add some lines to a testcase and forget about it. But so it happens that lctn requires server restart and compatible FS, forcing the testcase split, and note, that the lctn_0, lctn_2 testcases would be very short each, as there is no need to re-test the whole core functionality. Yes, probably read_free_rpl_tables would need the same so that we know that this filter does actually DTRT on different FS and lctn values, but this is not different from testing any other option combination?

Also the above is what I see the ideal for this particular testing, which is a prerequisite for keeping the same cross-platform functionality as the upstream server. We don't really need to implement this due to our limited supported platform set, but as we touch things, I believe it's good to have this in mind and work towards that direction to some extent.